### DISCUSSION TITLE
Renaming tunnels

### DISCUSSION THREAD

**Discussion by prlaba** [logisim-evolution/logisim-evolution#1440] [Q&A]
I use tunnels frequently in my circuits, not only to declutter my circuits but to make it easier to move parts of my circuits around.  Many of my circuits consist only of main components and tunnels connecting them together.

I find myself often renaming a tunnel to reflect changes in signal names.  But the process is tedious, as I have to rename each of the same-named tunnels individually.  And if I miss one, my circuit stops working and can be difficult to debug.

L-E lets me select multiple tunnels when renaming one of them, but when I do that all of the other selected tunnels are assigned the new name with a “-n” suffix added.  So what used to be a set of same-named tunnels are now separately named tunnels.

What’s the rationale for this behavior?  Wouldn’t you want *all* of the selected tunnels to be assigned the new name?

If you have to maintain the “-n” naming behavior, can you please consider providing an option that lets me assign the same name to all selected tunnels, without any “-n” suffix?

(FYI- The Digital simulator pops up a dialog box asking if you want to assign the name to all selected tunnels.)

Thanks,

**Comment by maehne** [comment]
Thanks @prlaba for the feedback concerning the renaming of tunnels. We agree that the current behaviour is non-intuitive. Your issue #1432 is already under investigation by @BFH-ktt1. However, we cannot commit to a date until when this will be fixed, as our bandwidth is limited. PRs addressing the issue are welcome!
